datalad_service/tasks/description.py

```diff
@@ -0,0 +1,27 @@
+import json
+import os
+from datalad_service.common.celery import dataset_task
+
+def edit_description(description, new_fields):
+    updated = description.copy()
+    updated.update(new_fields)
+    return updated
+
+@dataset_task
+def update_description(store, dataset, ds, description_fields):
+    if any (description_fields):
+        ds = ds if ds is not None else store.get_dataset(dataset)
```

Comment:
This should be simplified to always call store.get_dataset like the other tasks. The dataset object can never be passed as a Celery argument and should not be part of the signature here.